src/opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -342,73 +342,134 @@ Schedule::Schedule(const Deck& deck, const EclipseState& es, const std::optional
 
 namespace {
 
-class ScheduleLogger {
+class ScheduleLogger
+{
 public:
-    ScheduleLogger(bool restart_skip, const std::string& prefix_, const KeywordLocation& location)
-        : prefix(prefix_)
-        , current_file(location.filename)
+    enum class Stream
     {
-        if (restart_skip)
-            this->log_function = &OpmLog::note;
-        else
-            this->log_function = &OpmLog::info;
+        Info, Note, Debug,
+    };
+
+    ScheduleLogger(const Stream stream, const std::string& prefix, const KeywordLocation& location)
+        : stream_      {stream}
+        , prefix_      {prefix}
+        , current_file_{location.filename}
+    {
+        switch (this->stream_) {
+        case Stream::Info:
+            this->log_function_ = &OpmLog::info;
+            break;
+
+        case Stream::Note:
+            this->log_function_ = &OpmLog::note;
+            break;
+
+        case Stream::Debug:
+            this->log_function_ = &OpmLog::debug;
+            break;
+        }
```

Comment:
Wouldn't it be cleaner to pass a `std::function<...>` directly instead of the enum and store that as log_function`? Of coarse one would need a bool to indicate whether this is debug.

Reply:
Might be personal taste, though.